http/server.go

```diff
@@ -0,0 +1,132 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ipfs/go-log/v2"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/core"
+	"github.com/drand/drand/key"
+	"github.com/drand/drand/protobuf/drand"
+)
+
+// New creates an HTTP handler for the public Drand API
+func New(ctx context.Context, client drand.PublicClient) (http.Handler, error) {
```

Comment:
Why does the constructor needs a long lived context ? and not creates one per-request ?

Reply:
the constructor here was meant to allow for graceful shutdown of the full server in e.g. tests.
there's a sub-context created for each request scope.

Reply:
I would much prefer to have a Stop() / Close() method then, as now it forces to keep the http hanler + context around instead of just the handler, and context semantics is more for "requests" passing through the stacks.